.tekton/retag-image-task.yaml

```diff
@@ -0,0 +1,108 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: retag-image
+  namespace: rh-acs-tenant
+# TODO(ROX-22196): Refactor to a task bundle
+spec:
+  description: |
+    Copies input image to the desired repo with the desired tag.
+    Handles index images that reference per-arch images.
+    Copies attached `cosign` and `oras` artifacts.
+    Copies source images that are logically linked to the input image digest.
+  params:
+  - name: INPUT_IMAGE_REPO
+    description: Repository of the input image.
+    type: string
+  - name: INPUT_IMAGE_TAG
+    description: Tag of the input image.
+    type: string
+  - name: OUTPUT_IMAGE_REPO
+    description: Repository of the output image.
+    type: string
+  - name: OUTPUT_IMAGE_TAG
+    description: Repository of the output image.
+    type: string
+  results:
+  - name: IMAGE_DIGEST
+    description: Digest of the output image (will be the same as of the input one).
+  - name: IMAGE_URL
+    description: Image repository and tag of the output image.
+  - name: IMAGE_REF
+    description: Image reference of the output image containing both the repository, the tag and the digest.
+  steps:
+  - name: retag-image
+    image: quay.io/konflux-ci/release-service-utils:latest@sha256:0fdaefe6748962003e24f3faede4edf1abbea63112c6ad973654a0726aafb9b8
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+
+      SKOPEO_RETRIES=5
+
+      function copy_image() {
+        local -r input_url="${1}"
+        local -r output_url="${2}"
+
+        if [[ "${input_url}" == "${output_url}" ]]; then
+          >&2 echo "Error: Will not copy image ${input_url} into itself."
+          exit 5
+        fi
+
+        echo ">>> Copying image from ${input_url} ..."
+        echo ">>> ... to ${output_url} ..."
+
+        # cosign copies not just cosign artifacts but also images. It understands and copies index images.
+        # --force argument is needed to prevent the command failing when it runs into (partial) previous copies with
+        # errors like the following.
+        #   Error: image "quay.io/rhacs-eng/collector:sha256-4b509fdf27761150a5a5ce519f087bc6d69af9d886a63f67f70b1e293643ea19" already exists. Use `-f` to overwrite
+        cosign copy --force "${input_url}" "${output_url}"
+
+        oras cp --recursive "${input_url}" "${output_url}"
```

Comment:
Unfortunately, these don't retry. I noticed a `retry` command recently appeared in `quay.io/redhat-appstudio/build-trusted-artifacts` - https://github.com/konflux-ci/build-definitions/blob/3dd67af957e502463b56ea35cb18b7f0c766f158/task/push-dockerfile-oci-ta/0.1/push-dockerfile-oci-ta.yaml#L110 but it did not find its way to `quay.io/konflux-ci/release-service-utils` we use here.